newIDE/app/src/AssetStore/ProductPageHelper.js

```diff
@@ -269,3 +281,164 @@ export const getUserProductPurchaseUsageType = <
 
   return productPurchase.usageType;
 };
+
+export const PurchaseProductButtons = <
+  T: PrivateAssetPackListingData | PrivateGameTemplateListingData
+>({
+  productListingData,
+  selectedUsageType,
+  onUsageTypeChange,
+  simulateAppStoreProduct,
+  i18n,
+  isAlreadyReceived,
+  onClickBuy,
+  onClickBuyWithCredits,
+}: {|
+  productListingData: T,
+  selectedUsageType: string,
+  onUsageTypeChange: (usageType: string) => void,
+  simulateAppStoreProduct?: boolean,
+  i18n: I18nType,
+  isAlreadyReceived: boolean,
+  onClickBuy: () => Promise<void>,
+  onClickBuyWithCredits: () => Promise<void>,
+|}) => {
+  const { authenticated } = React.useContext(AuthenticatedUserContext);
+  const shouldUseOrSimulateAppStoreProduct =
+    simulateAppStoreProduct || shouldUseAppStoreProduct();
+  const productType = productListingData.productType.toLowerCase();
+
+  let creditPrice = productListingData.creditPrices.find(
+    price => price.usageType === selectedUsageType
+  );
+  if (!creditPrice) {
+    // We're probably switching from one product to another, and the usage type is not available.
+    // Let's reset it.
+    onUsageTypeChange(productListingData.prices[0].usageType);
+    creditPrice = productListingData.creditPrices.find(
+      price => price.usageType === productListingData.prices[0].usageType
+    );
+    if (!creditPrice) {
+      console.error('Unable to find a credit price for this product');
+      return null;
+    }
+  }
+
+  const formattedProductPriceText = formatProductPrice({
+    i18n,
+    usageType: selectedUsageType,
+    productListingData: productListingData,
+    plainText: true,
+  });
+
+  return shouldUseOrSimulateAppStoreProduct ? (
+    <LineStackLayout>
+      <RaisedButton
+        primary
+        label={<Trans>Buy for {formattedProductPriceText}</Trans>}
```

Comment:
Should it mention only the creditPrice.amount here?

Reply:
the function now either return:
- `$ 20`
- `2000 credits` 
Automatically based on the product and the environment. (Mentioning credits is actually better UX-wise)